### ISSUE TITLE
[BUG] ForecastingGridSearchCV with a ReducedRegressionForecaster does not display the right number of candidates

### ISSUE THREAD

**Issue by ngupta23** [alan-turing-institute/sktime#585]
**Describe the bug**

1. Using the `ForecastingGridSearchCV` with a `ReducedRegressionForecaster` does not display the right number of candidates
2. Eventually, the best_parameters_ for the underlying sklearn regressor are not accessible

**To Reproduce**
```python
y = load_airline()
y_train, y_test = temporal_train_test_split(y, test_size=36)

regressor_param_grid = {"n_estimators": [100, 200, 300]}
forecaster_param_grid = {
    "deseasonalise__sp": [6, 12, 24],
    "detrend__forecaster__degree": [1, 2],
    "forecaster__window_length": [10, 12, 15, 20, 25]
}

# create a tunnable regressor with GridSearchCV
regressor = GridSearchCV(RandomForestRegressor(random_state=42), param_grid=regressor_param_grid)
forecaster = TransformedTargetForecaster(
    [
        ("deseasonalise", Deseasonalizer(model="multiplicative")),  
        ("detrend", Detrender(forecaster=PolynomialTrendForecaster())),
        ("forecaster", ReducedRegressionForecaster(regressor=regressor, strategy="recursive")) 
    ]
)

cv = SlidingWindowSplitter(
    initial_window=int(len(y_train) * 0.5),
    start_with_window=True,
    window_length=36
)
gscv = ForecastingGridSearchCV(forecaster, cv=cv, param_grid=forecaster_param_grid, verbose=True)
%time gscv.fit(y_train)
```

```terminal
Fitting 72 folds for each of 30 candidates, totalling 2160 fits
Wall time: 3min 1s
```

**Expected behavior**

1. I was expecting to see 30 * 3 = 90 candidates. I think the count is ignoring the underlying parameters in sklearn's GridSearchCV.
2. Secondly, I tried to access the underlying random forest regressor best parameters but could not, even though the regressor itself could be accessed. Not sure if this is a bug.

```python
print(f"Best Params: {gscv.best_params_}")
print(f"Detrender Degree: {gscv.forecaster.named_steps['detrend'].forecaster.degree}")
print(f"Internal Regressor: {gscv.forecaster.named_steps['forecaster'].regressor}")  # Unable to print best_params_
```

```terminal
Best Params: {'deseasonalise__sp': 6, 'detrend__forecaster__degree': 1, 'forecaster__window_length': 25}
Detrender Degree: 1
Internal Regressor: GridSearchCV(estimator=RandomForestRegressor(random_state=42),
             param_grid={'n_estimators': [100, 200, 300]})
```

**Versions**
```
System:
    python: 3.6.12 |Anaconda, Inc.| (default, Sep  9 2020, 00:29:25) [MSC v.1916 64 bit (AMD64)]
executable: C:\Users\Nikhil\.conda\envs\sktime_dev\python.exe
   machine: Windows-10-10.0.18362-SP0

Python dependencies:
          pip: 20.3.3
   setuptools: 51.0.0.post20201207
      sklearn: 0.24.0
       sktime: 0.5.1
  statsmodels: 0.12.1
        numpy: 1.19.4
        scipy: 1.5.4
       Cython: 0.29.17
       pandas: 1.1.5
   matplotlib: 3.3.3
       joblib: 1.0.0
        numba: 0.52.0
     pmdarima: 1.8.0
      tsfresh: 0.17.0
```

**Comment by mloning** [comment]
Thanks @ngupta23 for raising this issue. 

The inner `GridSearchCV` is quite hidden but you can find it here: `gscv.best_forecaster_.steps_[-1][-1].regressor_.best_estimator_`.

**Comment by ngupta23** [comment]
Thanks @mloning. Yes, the candidates are correct if we only consider the outer folds, but was a little misleading. Maybe there could be a check for this and added to the total number of candidates (or even mentioned separately as outer and inner candidates). Just a thought. It threw me off initially since I was not sure if the hyper-parameters were being searched correctely for the Random Forest or not and that led me to the path of querying the inner regressor to check.

**Comment by mloning** [comment]
Combining them will add considerable complexity I think, but I agree it would be nice for users to see the entire grid.

**Comment by ngupta23** [comment]
Hi @mloning , My comment was only related to the reporting. I think the code flow can remain the same as you already have. 
If nothing else, maybe just a info print that the candidates are only for the outer loop would be good.

**Comment by mloning** [comment]
Or perhaps simply print the info from the inner loop, not sure why it is suppressed?

**Comment by ngupta23** [comment]
@mloning, Good point. I can enable this in the regrerssor instantiation. After I change it to the following, it starts showing up,  but now it is printing too much (1 for each of the outer candidates)

:)

```python
# create a tunnable regressor with GridSearchCV
regressor = GridSearchCV(
    RandomForestRegressor(random_state=42), param_grid=regressor_param_grid, verbose=1
)
```

```python
Fitting 72 folds for each of 30 candidates, totalling 2160 fits
Fitting 5 folds for each of 3 candidates, totalling 15 fits
Fitting 5 folds for each of 3 candidates, totalling 15 fits
Fitting 5 folds for each of 3 candidates, totalling 15 fits
Fitting 5 folds for each of 3 candidates, totalling 15 fits
...
```

**Comment by ngupta23** [comment]
And as I am printing these details, other questions are arising about the entire Grid Search process, but I will file a separate Issue for that to keep it clean.

**Comment by ngupta23** [comment]
A followup on the above output...

It looks like the scikit GridSearchCV is being fit using 5 folds using the "initial_window" data. But is this really used anywhere in the process? It does not look like it. Can you please confirm?

**Comment by mloning** [comment]
@ngupta23 what is not really used anywhere in the process?

**Comment by ngupta23** [comment]
@mloning, the 5 folds are only being created on the "initial_window", but the scores from those 5 folds are not being used anywhere since the scoring happens on the validation data and not in the "initial_window" data. Is this correct?

**Comment by mloning** [comment]
Yes I believe so - the best model is chosen on the initial training set and then we would score and update on the validation set (but as discussed updating isn't fully implemented yet).

**Comment by ngupta23** [comment]
@mloning, In that case, since the information from the 5 folds of the `GridSearchCV` is not being used anywhere, would it make sense to unravel the `ParamGrid` from `GridSearchCV` and combine it with the `ParamGrid` from `ForecastingGridSearchCV`. That way, we only fit on the initial window once (instead of 5 folds) for each set of candidate parameters and then evaluate on the "validation" set. This would be more efficient in terms of training time,

**Comment by ngupta23** [comment]
> Yes I believe so - the best model is chosen on the initial training set and then we would score and update on the validation set (but as discussed updating isn't fully implemented yet).

And just to clarify, it seems that only the regressor `best_params_` are selected in the "initial_window" but the Forecaster "best parameters" are selected using the "validation set". Is this understanding correct?

**Comment by mloning** [comment]
The regressor's best parameters are selected on the validation set of the (tabularized) initial window, where the validation set depends on the CV iterator you choose (e.g. TimeSeriesSplit).

**Comment by mloning** [comment]
I'll close this now. Please feel free to reopen if you want to give us more feedback. 

Related improvements based on the discussion here were merged in #739.
